lms/resources/_js_config/__init__.py

```diff
@@ -463,48 +470,70 @@ def _hypothesis_client(self):
         }
 
     def _groups(self):
-        if (
-            self._context.canvas_sections_enabled
-            or self._context.canvas_is_group_launch
-        ):
+        if self._context.canvas_sections_enabled or self._context.is_group_launch:
             return "$rpc:requestGroups"
         return [self._context.h_group.groupid(self._authority)]
 
     def _sync_api(self):
         if (
             not self._context.canvas_sections_enabled
-            and not self._context.canvas_is_group_launch
+            and not self._context.is_group_launch
         ):
             return None
 
         req = self._request
 
-        sync_api_config = {
-            "authUrl": req.route_url("canvas_api.oauth.authorize"),
-            "path": req.route_path("canvas_api.sync"),
-            "data": {
-                "lms": {
-                    "tool_consumer_instance_guid": req.params[
-                        "tool_consumer_instance_guid"
-                    ],
+        sync_api_config = {}
+        if self._context.is_canvas:
+            sync_api_config = {
+                "authUrl": req.route_url("canvas_api.oauth.authorize"),
+                "path": req.route_path("canvas_api.sync"),
+                "data": {
+                    "lms": {
+                        "tool_consumer_instance_guid": req.params[
+                            "tool_consumer_instance_guid"
+                        ],
+                    },
+                    "course": {
+                        "context_id": req.params["context_id"],
+                        "custom_canvas_course_id": req.params[
+                            "custom_canvas_course_id"
+                        ],
+                        "group_set": req.params.get("group_set"),
+                    },
+                    "group_info": {
+                        key: value
+                        for key, value in req.params.items()
+                        if key in GroupInfo.columns()
+                    },
                 },
-                "course": {
-                    "context_id": req.params["context_id"],
-                    "custom_canvas_course_id": req.params["custom_canvas_course_id"],
+            }
+
+            if "learner_canvas_user_id" in req.params:
+                sync_api_config["data"]["learner"] = {
+                    "canvas_user_id": req.params["learner_canvas_user_id"],
                     "group_set": req.params.get("group_set"),
+                }
+        if self._context.is_blackboard:
```

Comment:
I think we need to only include the sync API config **for Blackboard groups assignments**. Most Blackboard installations won't have the groups feature enabled because they won't have given us access to the Blackboard API, so we don't want to make any requests to the Blackboard API by default.

Reply:
Good point, I think the sync api itself also should check if it's a groups assignment too.